src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordEngine.cs

```diff
@@ -72,16 +77,48 @@ internal static class ConvertToRecordCommon
         }
 
-        private static async Task<Document> ConvertToPositionalRecordAsync(
+        private static async Task<Solution> ConvertToPositionalRecordAsync(
             Document document,
             INamedTypeSymbol type,
             ImmutableArray<PositionalParameterInfo> positionalParameterInfos,
             TypeDeclarationSyntax typeDeclaration,
             CancellationToken cancellationToken)
         {
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            using var _ = ArrayBuilder<MemberDeclarationSyntax>.GetInstance(out var modifiedMembers);
-            modifiedMembers.AddRange(typeDeclaration.Members);
+
+            // first see if we need to re-order our primary constructor parameters.
+            var propertiesToAssign = positionalParameterInfos.SelectAsArray(info => info.Symbol);
+            var primaryConstructor = typeDeclaration.Members
+                .OfType<ConstructorDeclarationSyntax>()
+
+                .FirstOrDefault(constructor =>
+                {
+                    var constructorSymbol = (IMethodSymbol)semanticModel
+                        .GetRequiredDeclaredSymbol(constructor, cancellationToken);
+                    var constructorOperation = (IConstructorBodyOperation)semanticModel
+                        .GetRequiredOperation(constructor, cancellationToken);
+                    // We want to make sure that each type in the parameter list corresponds
+                    // to exactly one positional parameter type, but they don't need to be in the same order.
+                    // We can't use something like set equality because some parameter types may be duplicate.
+                    // So, we order the types in a consistent way (by name) and then compare the lists of types.
+                    return constructorSymbol.Parameters.SelectAsArray(parameter => parameter.Type)
+                                    .OrderBy(type => type.Name)
+                            .SequenceEqual(propertiesToAssign.SelectAsArray(s => s.Type)
+                                    .OrderBy(type => type.Name),
+                                SymbolEqualityComparer.Default) &&
+                        // make sure that we do all the correct assignments. There may be multiple constructors
+                        // that meet the parameter condition but only one actually assigns all properties.
+                        // If successful, we set propertiesToAssign in the order of the parameters.
+                        ConvertToRecordHelpers.IsSimplePrimaryConstructor(
+                            constructorOperation, ref propertiesToAssign, constructorSymbol.Parameters);
+                });
+
+            var solutionEditor = new SolutionEditor(document.Project.Solution);
+            await RefactorInitializersAsync(type, solutionEditor, propertiesToAssign, cancellationToken)
+                .ConfigureAwait(false);
```

Comment:
Does this still need a comment this is being done first for any reason? Or did ordering not really matter in the end?